### ISSUE TITLE
Segmentation fault 11

### ISSUE THREAD

**Issue by atibaut** [IfcOpenShell/IfcOpenShell#637]
In the process of building with python3 build-all.py the process throws segmentation fault 11. I have checked all related  problems (https://github.com/IfcOpenShell/IfcOpenShell/issues?utf8=✓&q=segmentation+fault+11) but none of them applies to the problem because it happens when this line is executed:
`/git/IfcOpenShell/build/Darwin/x86_64/install/python-3.7.3/bin/python3 -c from __future__ import print_function; import inspect, ifcopenshell; print(inspect.getfile(ifcopenshell))`
The problem occurs with `import ifcopenshell`
Here is the complete log:

```
$ python3 build-all.py
This script fetches and builds IfcOpenShell and its dependencies

Script configuration:

* Target Architecture    = b'x86_64'
 - Whether 32-bit (i686) or 64-bit (x86_64) will be built.
* USE_OCCT               = True
 - Compiling against official Open Cascade
* Dependency Directory   = 
/git/IfcOpenShell/build/Darwin/x86_64
 - The directory where IfcOpenShell dependencies are installed.
* Build Config Type      = RelWithDebInfo
 - The used build configuration type for the dependencies.
   Defaults to RelWithDebInfo if not specified.
* IFCOS_NUM_BUILD_PROCS  = 9
 - How many compiler processes may be run in parallel.

Building: boost pcre swig libxml2 python occ json IfcParse OpenCOLLADA IfcGeom IfcGeomServer IfcOpenShell-Python IfcConvert
using command log file '/git/IfcOpenShell/build/Darwin/x86_64/logs/20190702.log'
Collecting dependencies:
Found existing cmake-3.14.5, skipping
Found existing pcre-8.43, skipping
Found existing swig, skipping
Found existing occt-7.3.0p3, skipping
Found existing libxml2-2.9.9, skipping
Found existing OpenCOLLADA, skipping
Downloading python-3.7.3...   
Configuring python-3.7.3...
Building python-3.7.3...   
Installing python-3.7.3... 
Installed python-3.7.3     

Found existing boost-1.59.0, skipping
Building IfcOpenShell:
Configuring executables...
Building executables...   
Configuring python 3.7.3 wrapper...
Building python 3.7.3 wrapper...   
----------------------------------------------------------------------
b''
----------------------------------------------------------------------
Traceback (most recent call last):
  File "build-all.py", line 785, in <module>
    module_dir = os.path.dirname(run([PYTHON_EXECUTABLE, "-c", "from __future__ import print_function; import inspect, ifcopenshell; print(inspect.getfile(ifcopenshell))"]))
  File "build-all.py", line 311, in run
    raise Exception("Command `%s` returned exit code %d" % (" ".join(cmds), proc.returncode))
Exception: Command `/git/IfcOpenShell/build/Darwin/x86_64/install/python-3.7.3/bin/python3 -c from __future__ import print_function; import inspect, ifcopenshell; print(inspect.getfile(ifcopenshell))` returned exit code -11
```
```
$ /git/IfcOpenShell/build/Darwin/x86_64/install/python-3.7.3/bin/python3
Python 3.7.3 (default, Jul  2 2019, 07:28:34) 
[Clang 10.0.1 (clang-1001.0.46.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from __future__ import print_function
>>> import inspect
>>> import ifcopenshell
Segmentation fault: 11
```

How can I solve the problem?

**Comment by aothms** [comment]
I don't have a lot of time this week. It seems there are more issues related to Python 3.7 on OSX, such as https://stackoverflow.com/questions/51726709/django-crash-segmentation-fault-11-on-macos

Can you run the process in `lldb` and get the stacktrace. Not an export at OSX debugging, but probably something like:

~~~
lldb /git/IfcOpenShell/build/Darwin/x86_64/install/python-3.7.3/bin/python3
~~~

and then just running the same commands in the repl

**Comment by atibaut** [comment]
Well, debugging with LLDB doesn't show the error, actually it ignores it:

```
$ lldb /git/IfcOpenShell/build/Darwin/x86_64/install/python-3.7.3/bin/python3
(lldb) target create "/git/IfcOpenShell/build/Darwin/x86_64/install/python-3.7.3/bin/python3"
Current executable set to '/git/IfcOpenShell/build/Darwin/x86_64/install/python-3.7.3/bin/python3' (x86_64).
(lldb) script 
Python Interactive Interpreter. To exit, type 'quit()', 'exit()' or Ctrl-D.
>>> from __future__ import print_function
>>> import inspect
>>> import ifcopenshell
>>> print(inspect.getfile(ifcopenshell))
/Library/Python/2.7/site-packages/ifcopenshell/__init__.pyc
>>> 
```

Therefore I have checked the crash report in `~/Library/Logs/DiagnosticReports/` which shows that there is an error in the _ifcopenshell_wrapper.so. I think this might be a bad memory reference:

```
Process:               python3.7 [17530]
Path:                  /Users/USER/*/python3.7
Identifier:            python3.7
Version:               ???
Code Type:             X86-64 (Native)
Parent Process:        bash [69006]
Responsible:           python3.7 [17530]
User ID:               501

Date/Time:             2019-07-03 22:05:01.858 +0200
OS Version:            Mac OS X 10.14.5 (18F132)
Report Version:        12
Anonymous UUID:        76AF52FA-69F9-718A-F0AE-F3F7323ED133

Sleep/Wake UUID:       84812D5D-DA19-47AC-86C2-3C240D570B2C

Time Awake Since Boot: 460000 seconds
Time Since Wake:       5300 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000008
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [17530]

VM Regions Near 0x8:
--> 
    __TEXT                 00000001003b9000-00000001005b0000 [ 2012K] r-x/rwx SM=COW  /Users/USER/*/*.7

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   _ifcopenshell_wrapper.so      	0x00000001012bf407 PyTuple_New + 263 (tupleobject.c:133)
1   _ifcopenshell_wrapper.so      	0x00000001012c2452 PyType_Ready + 354 (typeobject.c:5176)
2   _ifcopenshell_wrapper.so      	0x00000001012c23c4 PyType_Ready + 212 (typeobject.c:5155)
3   _ifcopenshell_wrapper.so      	0x0000000100b2eb89 PyInit__ifcopenshell_wrapper + 2953 (IfcPythonPYTHON_wrap.cxx:1959)
4   python3                       	0x00000001004d0f53 _PyImport_LoadDynamicModuleWithSpec + 579 (importdl.c:159)
5   python3                       	0x00000001004d09a3 _imp_create_dynamic + 355 (import.c:2158)
6   python3                       	0x00000001003da416 _PyMethodDef_RawFastCallDict + 534
7   python3                       	0x00000001003d9f08 PyCFunction_Call + 200 (call.c:582)
8   python3                       	0x00000001004ad266 _PyEval_EvalFrameDefault + 30198
9   python3                       	0x00000001004b1075 _PyEval_EvalCodeWithName + 2981
10  python3                       	0x00000001003d9c16 _PyFunction_FastCallKeywords + 230 (call.c:440)
11  python3                       	0x00000001004b0319 call_function + 809
12  python3                       	0x00000001004ace53 _PyEval_EvalFrameDefault + 29155 (ceval.c:3094)
13  python3                       	0x00000001003da14d function_code_fastcall + 253 (call.c:283)
14  python3                       	0x00000001004b0319 call_function + 809
15  python3                       	0x00000001004ace33 _PyEval_EvalFrameDefault + 29123 (ceval.c:3111)
16  python3                       	0x00000001003da14d function_code_fastcall + 253 (call.c:283)
17  python3                       	0x00000001004b0319 call_function + 809
18  python3                       	0x00000001004acefe _PyEval_EvalFrameDefault + 29326 (ceval.c:3125)
19  python3                       	0x00000001003da14d function_code_fastcall + 253 (call.c:283)
20  python3                       	0x00000001004b0319 call_function + 809
21  python3                       	0x00000001004acefe _PyEval_EvalFrameDefault + 29326 (ceval.c:3125)
22  python3                       	0x00000001003da14d function_code_fastcall + 253 (call.c:283)
23  python3                       	0x00000001004b0319 call_function + 809
24  python3                       	0x00000001004acefe _PyEval_EvalFrameDefault + 29326 (ceval.c:3125)
25  python3                       	0x00000001003da14d function_code_fastcall + 253 (call.c:283)
26  python3                       	0x00000001003dba9b object_vacall + 619 (call.c:1198)
27  python3                       	0x00000001003dbb8d _PyObject_CallMethodIdObjArgs + 173 (call.c:1248)
28  python3                       	0x00000001004cf8fa PyImport_ImportModuleLevelObject + 1658 (import.c:1646)
29  python3                       	0x00000001004a270b builtin___import__ + 139 (bltinmodule.c:312)
30  python3                       	0x00000001003d9ed4 PyCFunction_Call + 148 (call.c:753)
31  python3                       	0x00000001004ad266 _PyEval_EvalFrameDefault + 30198
32  python3                       	0x00000001004b1075 _PyEval_EvalCodeWithName + 2981
33  python3                       	0x00000001003d9c16 _PyFunction_FastCallKeywords + 230 (call.c:440)
34  python3                       	0x00000001004b0319 call_function + 809
35  python3                       	0x00000001004acefe _PyEval_EvalFrameDefault + 29326 (ceval.c:3125)
36  python3                       	0x00000001004b1075 _PyEval_EvalCodeWithName + 2981
37  python3                       	0x00000001003d9712 _PyFunction_FastCallDict + 450 (call.c:376)
38  python3                       	0x00000001003dba9b object_vacall + 619 (call.c:1198)
39  python3                       	0x00000001003dbb8d _PyObject_CallMethodIdObjArgs + 173 (call.c:1248)
40  python3                       	0x00000001004cf9eb PyImport_ImportModuleLevelObject + 1899 (import.c:1817)
41  python3                       	0x00000001004ab504 _PyEval_EvalFrameDefault + 22676 (ceval.c:4770)
42  python3                       	0x00000001004b1075 _PyEval_EvalCodeWithName + 2981
43  python3                       	0x00000001004a5bd0 PyEval_EvalCode + 48 (ceval.c:3959)
44  python3                       	0x00000001004a327f builtin_exec + 575 (bltinmodule.c:1079)
45  python3                       	0x00000001003da416 _PyMethodDef_RawFastCallDict + 534
46  python3                       	0x00000001003d9f08 PyCFunction_Call + 200 (call.c:582)
47  python3                       	0x00000001004ad266 _PyEval_EvalFrameDefault + 30198
48  python3                       	0x00000001004b1075 _PyEval_EvalCodeWithName + 2981
49  python3                       	0x00000001003d9c16 _PyFunction_FastCallKeywords + 230 (call.c:440)
50  python3                       	0x00000001004b0319 call_function + 809
51  python3                       	0x00000001004ace53 _PyEval_EvalFrameDefault + 29155 (ceval.c:3094)
52  python3                       	0x00000001003da14d function_code_fastcall + 253 (call.c:283)
53  python3                       	0x00000001004b0319 call_function + 809
54  python3                       	0x00000001004ace33 _PyEval_EvalFrameDefault + 29123 (ceval.c:3111)
55  python3                       	0x00000001003da14d function_code_fastcall + 253 (call.c:283)
56  python3                       	0x00000001004b0319 call_function + 809
57  python3                       	0x00000001004acefe _PyEval_EvalFrameDefault + 29326 (ceval.c:3125)
58  python3                       	0x00000001003da14d function_code_fastcall + 253 (call.c:283)
59  python3                       	0x00000001004b0319 call_function + 809
60  python3                       	0x00000001004acefe _PyEval_EvalFrameDefault + 29326 (ceval.c:3125)
61  python3                       	0x00000001003da14d function_code_fastcall + 253 (call.c:283)
62  python3                       	0x00000001003dba9b object_vacall + 619 (call.c:1198)
63  python3                       	0x00000001003dbb8d _PyObject_CallMethodIdObjArgs + 173 (call.c:1248)
64  python3                       	0x00000001004cf8fa PyImport_ImportModuleLevelObject + 1658 (import.c:1646)
65  python3                       	0x00000001004a270b builtin___import__ + 139 (bltinmodule.c:312)
66  python3                       	0x00000001003d9ed4 PyCFunction_Call + 148 (call.c:753)
67  python3                       	0x00000001004ad266 _PyEval_EvalFrameDefault + 30198
68  python3                       	0x00000001004b1075 _PyEval_EvalCodeWithName + 2981
69  python3                       	0x00000001003d9c16 _PyFunction_FastCallKeywords + 230 (call.c:440)
70  python3                       	0x00000001004b0319 call_function + 809
71  python3                       	0x00000001004acefe _PyEval_EvalFrameDefault + 29326 (ceval.c:3125)
72  python3                       	0x00000001004b1075 _PyEval_EvalCodeWithName + 2981
73  python3                       	0x00000001003d9712 _PyFunction_FastCallDict + 450 (call.c:376)
74  python3                       	0x00000001003dba9b object_vacall + 619 (call.c:1198)
75  python3                       	0x00000001003dbb8d _PyObject_CallMethodIdObjArgs + 173 (call.c:1248)
76  python3                       	0x00000001004cf9eb PyImport_ImportModuleLevelObject + 1899 (import.c:1817)
77  python3                       	0x00000001004ab504 _PyEval_EvalFrameDefault + 22676 (ceval.c:4770)
78  python3                       	0x00000001004b1075 _PyEval_EvalCodeWithName + 2981
79  python3                       	0x00000001004a5bd0 PyEval_EvalCode + 48 (ceval.c:3959)
80  python3                       	0x00000001004a327f builtin_exec + 575 (bltinmodule.c:1079)
81  python3                       	0x00000001003da416 _PyMethodDef_RawFastCallDict + 534
82  python3                       	0x00000001003d9f08 PyCFunction_Call + 200 (call.c:582)
83  python3                       	0x00000001004ad266 _PyEval_EvalFrameDefault + 30198
84  python3                       	0x00000001004b1075 _PyEval_EvalCodeWithName + 2981
85  python3                       	0x00000001003d9c16 _PyFunction_FastCallKeywords + 230 (call.c:440)
86  python3                       	0x00000001004b0319 call_function + 809
87  python3                       	0x00000001004ace53 _PyEval_EvalFrameDefault + 29155 (ceval.c:3094)
88  python3                       	0x00000001003da14d function_code_fastcall + 253 (call.c:283)
89  python3                       	0x00000001004b0319 call_function + 809
90  python3                       	0x00000001004ace33 _PyEval_EvalFrameDefault + 29123 (ceval.c:3111)
91  python3                       	0x00000001003da14d function_code_fastcall + 253 (call.c:283)
92  python3                       	0x00000001004b0319 call_function + 809
93  python3                       	0x00000001004acefe _PyEval_EvalFrameDefault + 29326 (ceval.c:3125)
94  python3                       	0x00000001003da14d function_code_fastcall + 253 (call.c:283)
95  python3                       	0x00000001004b0319 call_function + 809
96  python3                       	0x00000001004acefe _PyEval_EvalFrameDefault + 29326 (ceval.c:3125)
97  python3                       	0x00000001003da14d function_code_fastcall + 253 (call.c:283)
98  python3                       	0x00000001003dba9b object_vacall + 619 (call.c:1198)
99  python3                       	0x00000001003dbb8d _PyObject_CallMethodIdObjArgs + 173 (call.c:1248)
100 python3                       	0x00000001004cf8fa PyImport_ImportModuleLevelObject + 1658 (import.c:1646)
101 python3                       	0x00000001004ab504 _PyEval_EvalFrameDefault + 22676 (ceval.c:4770)
102 python3                       	0x00000001004b1075 _PyEval_EvalCodeWithName + 2981
103 python3                       	0x00000001004a5bd0 PyEval_EvalCode + 48 (ceval.c:3959)
104 python3                       	0x00000001004e8c7d PyRun_StringFlags + 141 (pythonrun.c:1035)
105 python3                       	0x00000001004e8bb5 PyRun_SimpleStringFlags + 69 (pythonrun.c:456)
106 python3                       	0x0000000100509f3b pymain_main + 5131 (main.c:385)
107 python3                       	0x000000010050b0ff _Py_UnixMain + 111 (main.c:3073)
108 libdyld.dylib                 	0x00007fff631403d5 start + 1

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x000000010256bc58  rbx: 0x000000010347c048  rcx: 0x000000010347c030  rdx: 0x0000000000000000
  rdi: 0x00000001024b5560  rsi: 0x0000000000000030  rbp: 0x00007ffeef841ce0  rsp: 0x00007ffeef841cd0
   r8: 0x000000000000000a   r9: 0xffffffff00000000  r10: 0x00007fff991f0f20  r11: 0x00000000000f34a8
  r12: 0x0000000100aebea0  r13: 0x0000000100b00470  r14: 0x0000000000000000  r15: 0x0000000100b00308
  rip: 0x00000001012bf407  rfl: 0x0000000000010286  cr2: 0x0000000000000008
  
Logical CPU:     6
Error Code:      0x00000004
Trap Number:     14


Binary Images:
       0x1003b9000 -        0x1005affff +python3 (0) <5FD0F822-69A9-3327-85C6-F51C5A0BFDC5> /Users/USER/*/python3
       0x100a0c000 -        0x100a0cff7 +_heapq.cpython-37m-darwin.so (0) <98C0FCCF-72B6-328F-B058-4FEB9053E34C> /Users/USER/*/_heapq.cpython-37m-darwin.so
       0x100a50000 -        0x100a50ff7 +_opcode.cpython-37m-darwin.so (0) <DBE311ED-2C49-3C77-A537-779B3DAFFB62> /Users/USER/*/_opcode.cpython-37m-darwin.so
       0x100b2b000 -        0x102448fff +_ifcopenshell_wrapper.so (0) <31377E1A-99D1-3159-AFB8-F1FDA8AC5CA2> /Users/USER/*/_ifcopenshell_wrapper.so
       0x10fe6a000 -        0x10fed46ef  dyld (655.1.1) <CE635DB2-D47E-3C05-A0A3-6BD982E7E750> /usr/lib/dyld
    0x7fff2b9a9000 -     0x7fff2b9aaff3  libLatin2Converter.dylib (73) <A4728CB6-A191-3563-97EB-7C34424D293B> /System/Library/CoreServices/Encodings/libLatin2Converter.dylib
    0x7fff371d7000 -     0x7fff3761bfff  com.apple.CoreFoundation (6.9 - 1575.12) <64C38BF2-1869-33C8-B1B6-DCA9C824C0E4> /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
    0x7fff5fce8000 -     0x7fff5fce9ffb  libDiagnosticMessagesClient.dylib (107) <A14D0819-0970-34CD-8680-80E4D7FE8C2C> /usr/lib/libDiagnosticMessagesClient.dylib
    0x7fff60127000 -     0x7fff60128ffb  libSystem.B.dylib (1252.250.1) <1A13E822-B59C-3A36-A2E4-9968149714F9> /usr/lib/libSystem.B.dylib
    0x7fff6036a000 -     0x7fff603bdff7  libc++.1.dylib (400.9.4) <9A60A190-6C34-339F-BB3D-AACE942009A4> /usr/lib/libc++.1.dylib
    0x7fff603be000 -     0x7fff603d3ff7  libc++abi.dylib (400.17) <38C09CED-9090-3719-90F3-04A2749F5428> /usr/lib/libc++abi.dylib
    0x7fff60e1a000 -     0x7fff6107bfff  libicucore.A.dylib (62135.0.1) <2B292561-798A-381C-AA41-90EE6549C7CF> /usr/lib/libicucore.A.dylib
    0x7fff6195e000 -     0x7fff620e3fdf  libobjc.A.dylib (756.2) <7C312627-43CB-3234-9324-4DEA92D59F50> /usr/lib/libobjc.A.dylib
    0x7fff627cd000 -     0x7fff627dfff7  libz.1.dylib (70.200.4) <B048FC1F-058F-3A08-A1FE-81D5308CB3E6> /usr/lib/libz.1.dylib
    0x7fff62fc3000 -     0x7fff62fc7ff3  libcache.dylib (81) <1987D1E1-DB11-3291-B12A-EBD55848E02D> /usr/lib/system/libcache.dylib
    0x7fff62fc8000 -     0x7fff62fd2ff3  libcommonCrypto.dylib (60118.250.2) <1765BB6E-6784-3653-B16B-CB839721DC9A> /usr/lib/system/libcommonCrypto.dylib
    0x7fff62fd3000 -     0x7fff62fdaff7  libcompiler_rt.dylib (63.4) <5212BA7B-B7EA-37B4-AF6E-AC4F507EDFB8> /usr/lib/system/libcompiler_rt.dylib
    0x7fff62fdb000 -     0x7fff62fe4ff7  libcopyfile.dylib (146.250.1) <98CD00CD-9B91-3B5C-A9DB-842638050FA8> /usr/lib/system/libcopyfile.dylib
    0x7fff62fe5000 -     0x7fff63069fc3  libcorecrypto.dylib (602.260.2) <BED0F385-D8A2-334A-BD7B-0673B5DABBC5> /usr/lib/system/libcorecrypto.dylib
    0x7fff630f0000 -     0x7fff63129ff7  libdispatch.dylib (1008.250.7) <26DF5B1E-A388-38EF-B73B-AF0E93FB02DB> /usr/lib/system/libdispatch.dylib
    0x7fff6312a000 -     0x7fff63156ff7  libdyld.dylib (655.1.1) <002418CC-AD11-3D10-865B-015591D24E6C> /usr/lib/system/libdyld.dylib
    0x7fff63157000 -     0x7fff63157ffb  libkeymgr.dylib (30) <0D0F9CA2-8D5A-3273-8723-59987B5827F2> /usr/lib/system/libkeymgr.dylib
    0x7fff63165000 -     0x7fff63165ff7  liblaunch.dylib (1336.261.2) <BD5CDB35-2BB4-349A-8D14-78E387563F89> /usr/lib/system/liblaunch.dylib
    0x7fff63166000 -     0x7fff6316bfff  libmacho.dylib (927.0.2) <3C5C9024-45FD-38C2-B01A-07A322966063> /usr/lib/system/libmacho.dylib
    0x7fff6316c000 -     0x7fff6316effb  libquarantine.dylib (86.220.1) <6D0BC770-7348-3608-9254-F7FFBD347634> /usr/lib/system/libquarantine.dylib
    0x7fff6316f000 -     0x7fff63170ff7  libremovefile.dylib (45.200.2) <9FBEB2FF-EEBE-31BC-BCFC-C71F8D0E99B6> /usr/lib/system/libremovefile.dylib
    0x7fff63171000 -     0x7fff63188ff3  libsystem_asl.dylib (356.200.4) <A62A7249-38B8-33FA-9875-F1852590796C> /usr/lib/system/libsystem_asl.dylib
    0x7fff63189000 -     0x7fff63189ff7  libsystem_blocks.dylib (73) <A453E8EE-860D-3CED-B5DC-BE54E9DB4348> /usr/lib/system/libsystem_blocks.dylib
    0x7fff6318a000 -     0x7fff63211fff  libsystem_c.dylib (1272.250.1) <7EDACF78-2FA3-35B8-B051-D70475A35117> /usr/lib/system/libsystem_c.dylib
    0x7fff63212000 -     0x7fff63215ffb  libsystem_configuration.dylib (963.260.1) <C7D2BA09-8D5B-3393-8D84-304D76C55DEF> /usr/lib/system/libsystem_configuration.dylib
    0x7fff63216000 -     0x7fff63219ff7  libsystem_coreservices.dylib (66) <719F75A4-74C5-3BA6-A09E-0C5A3E5889D7> /usr/lib/system/libsystem_coreservices.dylib
    0x7fff6321a000 -     0x7fff63220fff  libsystem_darwin.dylib (1272.250.1) <EC9B39A5-9592-3577-8997-7DC721D20D8C> /usr/lib/system/libsystem_darwin.dylib
    0x7fff63221000 -     0x7fff63227ff7  libsystem_dnssd.dylib (878.260.1) <64B79B01-B1B0-3C7E-87A1-023B71843F1F> /usr/lib/system/libsystem_dnssd.dylib
    0x7fff63228000 -     0x7fff63273ffb  libsystem_info.dylib (517.200.9) <D09D5AE0-2FDC-3A6D-93EC-729F931B1457> /usr/lib/system/libsystem_info.dylib
    0x7fff63274000 -     0x7fff6329cff7  libsystem_kernel.dylib (4903.261.4) <7B0F52C4-4CC9-3282-8457-C18C6FE1B99E> /usr/lib/system/libsystem_kernel.dylib
    0x7fff6329d000 -     0x7fff632e8ff7  libsystem_m.dylib (3158.200.7) <F19B6DB7-014F-3820-831F-389CCDA06EF6> /usr/lib/system/libsystem_m.dylib
    0x7fff632e9000 -     0x7fff63313fff  libsystem_malloc.dylib (166.261.1) <4D1CEF38-DE32-3ECF-9F70-0ADDD5C8775F> /usr/lib/system/libsystem_malloc.dylib
    0x7fff63314000 -     0x7fff6331eff7  libsystem_networkextension.dylib (767.250.2) <FF06F13A-AEFE-3A27-A073-910EF78AEA36> /usr/lib/system/libsystem_networkextension.dylib
    0x7fff6331f000 -     0x7fff63326fff  libsystem_notify.dylib (172.200.21) <145B5CFC-CF73-33CE-BD3D-E8DDE268FFDE> /usr/lib/system/libsystem_notify.dylib
    0x7fff63327000 -     0x7fff63330fef  libsystem_platform.dylib (177.250.1) <16949870-9152-3B0D-9EF0-40FAF84B0F1E> /usr/lib/system/libsystem_platform.dylib
    0x7fff63331000 -     0x7fff6333bff7  libsystem_pthread.dylib (330.250.2) <2D5C08FF-484F-3D59-9132-CE1DCB3F76D7> /usr/lib/system/libsystem_pthread.dylib
    0x7fff6333c000 -     0x7fff6333fff7  libsystem_sandbox.dylib (851.260.2) <9CACC5F5-3859-3A1F-BCC6-96DDD4A556A8> /usr/lib/system/libsystem_sandbox.dylib
    0x7fff63340000 -     0x7fff63342ff3  libsystem_secinit.dylib (30.260.2) <88F3CE61-5FAD-3CDA-AA13-B5B18B3EBD26> /usr/lib/system/libsystem_secinit.dylib
    0x7fff63343000 -     0x7fff6334aff3  libsystem_symptoms.dylib (820.267.1) <03F1C2DD-0F5A-3D9D-88F6-B26C0F94EB52> /usr/lib/system/libsystem_symptoms.dylib
    0x7fff6334b000 -     0x7fff63360fff  libsystem_trace.dylib (906.260.1) <100013AE-0443-3CF0-AC17-5D934608B60B> /usr/lib/system/libsystem_trace.dylib
    0x7fff63362000 -     0x7fff63367ffb  libunwind.dylib (35.4) <24A97A67-F017-3CFC-B0D0-6BD0224B1336> /usr/lib/system/libunwind.dylib
    0x7fff63368000 -     0x7fff63397fff  libxpc.dylib (1336.261.2) <9A0FFA79-082F-3293-BF49-63976B073B74> /usr/lib/system/libxpc.dylib

```

**Comment by aothms** [comment]
Thanks. That call stack is exactly what I meant. The error is in
`PyInit__ifcopenshell_wrapper`.
That is code generated by Swig. I wonder whether this is actually related
to ifcopenshell or could be replicated in any other module. If you have
time I'd be interested to know if a simple example swig module does work
when generated on your system.

Sent from a mobile device. Excuse my brevity. Kind regards, Thomas

Op wo 3 jul. 2019 22:41 schreef Andrej Tibaut <notifications@github.com>:

**Comment by atibaut** [comment]
@aothms I went thru the Swig example (https://www.expobrain.net/2011/01/23/swig-tutorial-for-mac-os-x/) and was able to reproduce it without error. I used the Swing Version: 4.0.0 (27 Apr 2019).

```
$ python3
Python 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 16:52:21) 
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import example
>>> example.fact(10)
3628800
>>> example.my_mod(14,4)
2
>>> example.get_time()
'Sat Jul  6 00:05:14 2019\n'
```

Any other hint?

**Comment by aothms** [comment]
Thanks! But if you use 3.0.12?

Sent from a mobile device. Excuse my brevity. Kind regards, Thomas

Op za 6 jul. 2019 00:23 schreef Andrej Tibaut <notifications@github.com>:

**Comment by atibaut** [comment]
@aothms I tested it also with the swig 3.0.12 and the build process resulted in the same error:

```
This script fetches and builds IfcOpenShell and its dependencies

Script configuration:

* Target Architecture    = b'x86_64'
 - Whether 32-bit (i686) or 64-bit (x86_64) will be built.
* USE_OCCT               = True
 - Compiling against official Open Cascade
* Dependency Directory   = /Users/andrejt/Research/repositories/git/IfcOpenShell/build/Darwin/x86_64
 - The directory where IfcOpenShell dependencies are installed.
* Build Config Type      = RelWithDebInfo
 - The used build configuration type for the dependencies.
   Defaults to RelWithDebInfo if not specified.
* IFCOS_NUM_BUILD_PROCS  = 9
 - How many compiler processes may be run in parallel.

Building: boost python pcre swig json libxml2 occ IfcParse OpenCOLLADA IfcGeom IfcGeomServer IfcOpenShell-Python IfcConvert
using command log file '/Users/andrejt/Research/repositories/git/IfcOpenShell/build/Darwin/x86_64/logs/20190709.log'
Collecting dependencies:
Found existing cmake-3.14.5, skipping
Found existing pcre-8.43, skipping
Found existing swig, skipping
Found existing occt-7.3.0p3, skipping
Found existing libxml2-2.9.9, skipping
Found existing OpenCOLLADA, skipping
Found existing python-3.7.3, skipping
Found existing boost-1.59.0, skipping
Building IfcOpenShell:
Configuring executables...
Building executables...   
Configuring python 3.7.3 wrapper...
Building python 3.7.3 wrapper...   
----------------------------------------------------------------------
b''
----------------------------------------------------------------------
Traceback (most recent call last):
  File "build-all.py", line 785, in <module>
    module_dir = os.path.dirname(run([PYTHON_EXECUTABLE, "-c", "from __future__ import print_function; import inspect, ifcopenshell; print(inspect.getfile(ifcopenshell))"]))
  File "build-all.py", line 311, in run
    raise Exception("Command `%s` returned exit code %d" % (" ".join(cmds), proc.returncode))
Exception: Command `/Users/andrejt/Research/repositories/git/IfcOpenShell/build/Darwin/x86_64/install/python-3.7.3/bin/python3 -c from __future__ import print_function; import inspect, ifcopenshell; print(inspect.getfile(ifcopenshell))` returned exit code -11
```

This is the call stack:

```
Process:               python3.7 [16961]
Path:                  /Users/USER/*/python3.7
Identifier:            python3.7
Version:               ???
Code Type:             X86-64 (Native)
Parent Process:        ??? [15934]
Responsible:           python3.7 [16961]
User ID:               501

Date/Time:             2019-07-09 07:42:49.928 +0200
OS Version:            Mac OS X 10.14.5 (18F132)
Report Version:        12
Anonymous UUID:        76AF52FA-69F9-718A-F0AE-F3F7323ED133

Sleep/Wake UUID:       B96C30B2-BCBA-442A-91BF-C04E3476231A

Time Awake Since Boot: 15000 seconds
Time Since Wake:       14000 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000010
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [16961]

VM Regions Near 0x10:
--> 
    __TEXT                 0000000102f1a000-0000000103111000 [ 2012K] r-x/rwx SM=COW  /Users/USER/*/*.7

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   _ifcopenshell_wrapper.so      	0x00000001040c136b PyModule_Create2 + 27 (moduleobject.c:176)
1   _ifcopenshell_wrapper.so      	0x000000010393e6dc PyInit__ifcopenshell_wrapper + 76 (IfcPythonPYTHON_wrap.cxx:23087)
2   python3                       	0x0000000103031f53 _PyImport_LoadDynamicModuleWithSpec + 579 (importdl.c:159)
3   python3                       	0x00000001030319a3 _imp_create_dynamic + 355 (import.c:2158)
4   python3                       	0x0000000102f3b416 _PyMethodDef_RawFastCallDict + 534
5   python3                       	0x0000000102f3af08 PyCFunction_Call + 200 (call.c:582)
6   python3                       	0x000000010300e266 _PyEval_EvalFrameDefault + 30198
7   python3                       	0x0000000103012075 _PyEval_EvalCodeWithName + 2981
8   python3                       	0x0000000102f3ac16 _PyFunction_FastCallKeywords + 230 (call.c:440)
9   python3                       	0x0000000103011319 call_function + 809
10  python3                       	0x000000010300de53 _PyEval_EvalFrameDefault + 29155 (ceval.c:3094)
11  python3                       	0x0000000102f3b14d function_code_fastcall + 253 (call.c:283)
12  python3                       	0x0000000103011319 call_function + 809
13  python3                       	0x000000010300de33 _PyEval_EvalFrameDefault + 29123 (ceval.c:3111)
14  python3                       	0x0000000102f3b14d function_code_fastcall + 253 (call.c:283)
15  python3                       	0x0000000103011319 call_function + 809
16  python3                       	0x000000010300defe _PyEval_EvalFrameDefault + 29326 (ceval.c:3125)
17  python3                       	0x0000000102f3b14d function_code_fastcall + 253 (call.c:283)
18  python3                       	0x0000000103011319 call_function + 809
19  python3                       	0x000000010300defe _PyEval_EvalFrameDefault + 29326 (ceval.c:3125)
20  python3                       	0x0000000102f3b14d function_code_fastcall + 253 (call.c:283)
21  python3                       	0x0000000103011319 call_function + 809
22  python3                       	0x000000010300defe _PyEval_EvalFrameDefault + 29326 (ceval.c:3125)
23  python3                       	0x0000000102f3b14d function_code_fastcall + 253 (call.c:283)
24  python3                       	0x0000000103011319 call_function + 809
25  python3                       	0x000000010300defe _PyEval_EvalFrameDefault + 29326 (ceval.c:3125)
26  python3                       	0x0000000103012075 _PyEval_EvalCodeWithName + 2981
27  python3                       	0x0000000102f3ac16 _PyFunction_FastCallKeywords + 230 (call.c:440)
28  python3                       	0x0000000103011319 call_function + 809
29  python3                       	0x000000010300de53 _PyEval_EvalFrameDefault + 29155 (ceval.c:3094)
30  python3                       	0x0000000103012075 _PyEval_EvalCodeWithName + 2981
31  python3                       	0x0000000102f3ac16 _PyFunction_FastCallKeywords + 230 (call.c:440)
32  python3                       	0x0000000103011319 call_function + 809
33  python3                       	0x000000010300de53 _PyEval_EvalFrameDefault + 29155 (ceval.c:3094)
34  python3                       	0x0000000102f3b14d function_code_fastcall + 253 (call.c:283)
35  python3                       	0x0000000103011319 call_function + 809
36  python3                       	0x000000010300defe _PyEval_EvalFrameDefault + 29326 (ceval.c:3125)
37  python3                       	0x0000000103012075 _PyEval_EvalCodeWithName + 2981
38  python3                       	0x0000000103006bd0 PyEval_EvalCode + 48 (ceval.c:3959)
39  python3                       	0x000000010300427f builtin_exec + 575 (bltinmodule.c:1079)
40  python3                       	0x0000000102f3b416 _PyMethodDef_RawFastCallDict + 534
41  python3                       	0x0000000102f3af08 PyCFunction_Call + 200 (call.c:582)
42  python3                       	0x000000010300e266 _PyEval_EvalFrameDefault + 30198
43  python3                       	0x0000000103012075 _PyEval_EvalCodeWithName + 2981
44  python3                       	0x0000000102f3ac16 _PyFunction_FastCallKeywords + 230 (call.c:440)
45  python3                       	0x0000000103011319 call_function + 809
46  python3                       	0x000000010300de53 _PyEval_EvalFrameDefault + 29155 (ceval.c:3094)
47  python3                       	0x0000000102f3b14d function_code_fastcall + 253 (call.c:283)
48  python3                       	0x0000000103011319 call_function + 809
49  python3                       	0x000000010300de33 _PyEval_EvalFrameDefault + 29123 (ceval.c:3111)
50  python3                       	0x0000000102f3b14d function_code_fastcall + 253 (call.c:283)
51  python3                       	0x0000000103011319 call_function + 809
52  python3                       	0x000000010300defe _PyEval_EvalFrameDefault + 29326 (ceval.c:3125)
53  python3                       	0x0000000102f3b14d function_code_fastcall + 253 (call.c:283)
54  python3                       	0x0000000103011319 call_function + 809
55  python3                       	0x000000010300defe _PyEval_EvalFrameDefault + 29326 (ceval.c:3125)
56  python3                       	0x0000000102f3b14d function_code_fastcall + 253 (call.c:283)
57  python3                       	0x0000000102f3ca9b object_vacall + 619 (call.c:1198)
58  python3                       	0x0000000102f3cb8d _PyObject_CallMethodIdObjArgs + 173 (call.c:1248)
59  python3                       	0x00000001030308fa PyImport_ImportModuleLevelObject + 1658 (import.c:1646)
60  python3                       	0x000000010300370b builtin___import__ + 139 (bltinmodule.c:312)
61  python3                       	0x0000000102f3aed4 PyCFunction_Call + 148 (call.c:753)
62  python3                       	0x000000010300e266 _PyEval_EvalFrameDefault + 30198
63  python3                       	0x0000000103012075 _PyEval_EvalCodeWithName + 2981
64  python3                       	0x0000000102f3ac16 _PyFunction_FastCallKeywords + 230 (call.c:440)
65  python3                       	0x0000000103011319 call_function + 809
66  python3                       	0x000000010300defe _PyEval_EvalFrameDefault + 29326 (ceval.c:3125)
67  python3                       	0x0000000103012075 _PyEval_EvalCodeWithName + 2981
68  python3                       	0x0000000102f3a712 _PyFunction_FastCallDict + 450 (call.c:376)
69  python3                       	0x0000000102f3ca9b object_vacall + 619 (call.c:1198)
70  python3                       	0x0000000102f3cb8d _PyObject_CallMethodIdObjArgs + 173 (call.c:1248)
71  python3                       	0x00000001030309eb PyImport_ImportModuleLevelObject + 1899 (import.c:1817)
72  python3                       	0x000000010300c504 _PyEval_EvalFrameDefault + 22676 (ceval.c:4770)
73  python3                       	0x0000000103012075 _PyEval_EvalCodeWithName + 2981
74  python3                       	0x0000000103006bd0 PyEval_EvalCode + 48 (ceval.c:3959)
75  python3                       	0x000000010300427f builtin_exec + 575 (bltinmodule.c:1079)
76  python3                       	0x0000000102f3b416 _PyMethodDef_RawFastCallDict + 534
77  python3                       	0x0000000102f3af08 PyCFunction_Call + 200 (call.c:582)
78  python3                       	0x000000010300e266 _PyEval_EvalFrameDefault + 30198
79  python3                       	0x0000000103012075 _PyEval_EvalCodeWithName + 2981
80  python3                       	0x0000000102f3ac16 _PyFunction_FastCallKeywords + 230 (call.c:440)
81  python3                       	0x0000000103011319 call_function + 809
82  python3                       	0x000000010300de53 _PyEval_EvalFrameDefault + 29155 (ceval.c:3094)
83  python3                       	0x0000000102f3b14d function_code_fastcall + 253 (call.c:283)
84  python3                       	0x0000000103011319 call_function + 809
85  python3                       	0x000000010300de33 _PyEval_EvalFrameDefault + 29123 (ceval.c:3111)
86  python3                       	0x0000000102f3b14d function_code_fastcall + 253 (call.c:283)
87  python3                       	0x0000000103011319 call_function + 809
88  python3                       	0x000000010300defe _PyEval_EvalFrameDefault + 29326 (ceval.c:3125)
89  python3                       	0x0000000102f3b14d function_code_fastcall + 253 (call.c:283)
90  python3                       	0x0000000103011319 call_function + 809
91  python3                       	0x000000010300defe _PyEval_EvalFrameDefault + 29326 (ceval.c:3125)
92  python3                       	0x0000000102f3b14d function_code_fastcall + 253 (call.c:283)
93  python3                       	0x0000000102f3ca9b object_vacall + 619 (call.c:1198)
94  python3                       	0x0000000102f3cb8d _PyObject_CallMethodIdObjArgs + 173 (call.c:1248)
95  python3                       	0x00000001030308fa PyImport_ImportModuleLevelObject + 1658 (import.c:1646)
96  python3                       	0x000000010300c504 _PyEval_EvalFrameDefault + 22676 (ceval.c:4770)
97  python3                       	0x0000000103012075 _PyEval_EvalCodeWithName + 2981
98  python3                       	0x0000000103006bd0 PyEval_EvalCode + 48 (ceval.c:3959)
99  python3                       	0x0000000103049c7d PyRun_StringFlags + 141 (pythonrun.c:1035)
100 python3                       	0x0000000103049bb5 PyRun_SimpleStringFlags + 69 (pythonrun.c:456)
101 python3                       	0x000000010306af3b pymain_main + 5131 (main.c:385)
102 python3                       	0x000000010306c0ff _Py_UnixMain + 111 (main.c:3073)
103 libdyld.dylib                 	0x00007fff7a1313d5 start + 1

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x0000000000000000  rbx: 0x00000001052aa030  rcx: 0x000000000000000f  rdx: 0x000000010352cb20
  rdi: 0x00000001052aa030  rsi: 0x00000000000003f5  rbp: 0x00007ffeecce0f00  rsp: 0x00007ffeecce0ef0
   r8: 0x0000000000000000   r9: 0x0000000000000000  r10: 0x0000000000000000  r11: 0x0000000000000000
  r12: 0x000000010361ed98  r13: 0x000000010367eac8  r14: 0x00000000000003f5  r15: 0x00000001039162d0
  rip: 0x00000001040c136b  rfl: 0x0000000000010202  cr2: 0x0000000000000010
  
Logical CPU:     4
Error Code:      0x00000004
Trap Number:     14


Binary Images:
       0x102f1a000 -        0x103110fff +python3 (0) <5FD0F822-69A9-3327-85C6-F51C5A0BFDC5> /Users/USER/*/python3
       0x1034ec000 -        0x1034ecff7 +_heapq.cpython-37m-darwin.so (0) <98C0FCCF-72B6-328F-B058-4FEB9053E34C> /Users/USER/*/_heapq.cpython-37m-darwin.so
       0x103530000 -        0x103530ff7 +_opcode.cpython-37m-darwin.so (0) <DBE311ED-2C49-3C77-A537-779B3DAFFB62> /Users/USER/*/_opcode.cpython-37m-darwin.so
       0x10393b000 -        0x10525cfff +_ifcopenshell_wrapper.so (0) <EC79EA3B-BCE3-3E2E-8148-41227E2ED74A> /Users/USER/*/_ifcopenshell_wrapper.so
       0x108460000 -        0x1084ca6ef  dyld (655.1.1) <CE635DB2-D47E-3C05-A0A3-6BD982E7E750> /usr/lib/dyld
    0x7fff4299a000 -     0x7fff4299bff3  libLatin2Converter.dylib (73) <A4728CB6-A191-3563-97EB-7C34424D293B> /System/Library/CoreServices/Encodings/libLatin2Converter.dylib
    0x7fff4e1c8000 -     0x7fff4e60cfff  com.apple.CoreFoundation (6.9 - 1575.12) <64C38BF2-1869-33C8-B1B6-DCA9C824C0E4> /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
    0x7fff76cd9000 -     0x7fff76cdaffb  libDiagnosticMessagesClient.dylib (107) <A14D0819-0970-34CD-8680-80E4D7FE8C2C> /usr/lib/libDiagnosticMessagesClient.dylib
    0x7fff77118000 -     0x7fff77119ffb  libSystem.B.dylib (1252.250.1) <1A13E822-B59C-3A36-A2E4-9968149714F9> /usr/lib/libSystem.B.dylib
    0x7fff7735b000 -     0x7fff773aeff7  libc++.1.dylib (400.9.4) <9A60A190-6C34-339F-BB3D-AACE942009A4> /usr/lib/libc++.1.dylib
    0x7fff773af000 -     0x7fff773c4ff7  libc++abi.dylib (400.17) <38C09CED-9090-3719-90F3-04A2749F5428> /usr/lib/libc++abi.dylib
    0x7fff77e0b000 -     0x7fff7806cfff  libicucore.A.dylib (62135.0.1) <2B292561-798A-381C-AA41-90EE6549C7CF> /usr/lib/libicucore.A.dylib
    0x7fff7894f000 -     0x7fff790d4fdf  libobjc.A.dylib (756.2) <7C312627-43CB-3234-9324-4DEA92D59F50> /usr/lib/libobjc.A.dylib
    0x7fff797be000 -     0x7fff797d0ff7  libz.1.dylib (70.200.4) <B048FC1F-058F-3A08-A1FE-81D5308CB3E6> /usr/lib/libz.1.dylib
    0x7fff79fb4000 -     0x7fff79fb8ff3  libcache.dylib (81) <1987D1E1-DB11-3291-B12A-EBD55848E02D> /usr/lib/system/libcache.dylib
    0x7fff79fb9000 -     0x7fff79fc3ff3  libcommonCrypto.dylib (60118.250.2) <1765BB6E-6784-3653-B16B-CB839721DC9A> /usr/lib/system/libcommonCrypto.dylib
    0x7fff79fc4000 -     0x7fff79fcbff7  libcompiler_rt.dylib (63.4) <5212BA7B-B7EA-37B4-AF6E-AC4F507EDFB8> /usr/lib/system/libcompiler_rt.dylib
    0x7fff79fcc000 -     0x7fff79fd5ff7  libcopyfile.dylib (146.250.1) <98CD00CD-9B91-3B5C-A9DB-842638050FA8> /usr/lib/system/libcopyfile.dylib
    0x7fff79fd6000 -     0x7fff7a05afc3  libcorecrypto.dylib (602.260.2) <BED0F385-D8A2-334A-BD7B-0673B5DABBC5> /usr/lib/system/libcorecrypto.dylib
    0x7fff7a0e1000 -     0x7fff7a11aff7  libdispatch.dylib (1008.250.7) <26DF5B1E-A388-38EF-B73B-AF0E93FB02DB> /usr/lib/system/libdispatch.dylib
    0x7fff7a11b000 -     0x7fff7a147ff7  libdyld.dylib (655.1.1) <002418CC-AD11-3D10-865B-015591D24E6C> /usr/lib/system/libdyld.dylib
    0x7fff7a148000 -     0x7fff7a148ffb  libkeymgr.dylib (30) <0D0F9CA2-8D5A-3273-8723-59987B5827F2> /usr/lib/system/libkeymgr.dylib
    0x7fff7a156000 -     0x7fff7a156ff7  liblaunch.dylib (1336.261.2) <BD5CDB35-2BB4-349A-8D14-78E387563F89> /usr/lib/system/liblaunch.dylib
    0x7fff7a157000 -     0x7fff7a15cfff  libmacho.dylib (927.0.2) <3C5C9024-45FD-38C2-B01A-07A322966063> /usr/lib/system/libmacho.dylib
    0x7fff7a15d000 -     0x7fff7a15fffb  libquarantine.dylib (86.220.1) <6D0BC770-7348-3608-9254-F7FFBD347634> /usr/lib/system/libquarantine.dylib
    0x7fff7a160000 -     0x7fff7a161ff7  libremovefile.dylib (45.200.2) <9FBEB2FF-EEBE-31BC-BCFC-C71F8D0E99B6> /usr/lib/system/libremovefile.dylib
    0x7fff7a162000 -     0x7fff7a179ff3  libsystem_asl.dylib (356.200.4) <A62A7249-38B8-33FA-9875-F1852590796C> /usr/lib/system/libsystem_asl.dylib
    0x7fff7a17a000 -     0x7fff7a17aff7  libsystem_blocks.dylib (73) <A453E8EE-860D-3CED-B5DC-BE54E9DB4348> /usr/lib/system/libsystem_blocks.dylib
    0x7fff7a17b000 -     0x7fff7a202fff  libsystem_c.dylib (1272.250.1) <7EDACF78-2FA3-35B8-B051-D70475A35117> /usr/lib/system/libsystem_c.dylib
    0x7fff7a203000 -     0x7fff7a206ffb  libsystem_configuration.dylib (963.260.1) <C7D2BA09-8D5B-3393-8D84-304D76C55DEF> /usr/lib/system/libsystem_configuration.dylib
    0x7fff7a207000 -     0x7fff7a20aff7  libsystem_coreservices.dylib (66) <719F75A4-74C5-3BA6-A09E-0C5A3E5889D7> /usr/lib/system/libsystem_coreservices.dylib
    0x7fff7a20b000 -     0x7fff7a211fff  libsystem_darwin.dylib (1272.250.1) <EC9B39A5-9592-3577-8997-7DC721D20D8C> /usr/lib/system/libsystem_darwin.dylib
    0x7fff7a212000 -     0x7fff7a218ff7  libsystem_dnssd.dylib (878.260.1) <64B79B01-B1B0-3C7E-87A1-023B71843F1F> /usr/lib/system/libsystem_dnssd.dylib
    0x7fff7a219000 -     0x7fff7a264ffb  libsystem_info.dylib (517.200.9) <D09D5AE0-2FDC-3A6D-93EC-729F931B1457> /usr/lib/system/libsystem_info.dylib
    0x7fff7a265000 -     0x7fff7a28dff7  libsystem_kernel.dylib (4903.261.4) <7B0F52C4-4CC9-3282-8457-C18C6FE1B99E> /usr/lib/system/libsystem_kernel.dylib
    0x7fff7a28e000 -     0x7fff7a2d9ff7  libsystem_m.dylib (3158.200.7) <F19B6DB7-014F-3820-831F-389CCDA06EF6> /usr/lib/system/libsystem_m.dylib
    0x7fff7a2da000 -     0x7fff7a304fff  libsystem_malloc.dylib (166.261.1) <4D1CEF38-DE32-3ECF-9F70-0ADDD5C8775F> /usr/lib/system/libsystem_malloc.dylib
    0x7fff7a305000 -     0x7fff7a30fff7  libsystem_networkextension.dylib (767.250.2) <FF06F13A-AEFE-3A27-A073-910EF78AEA36> /usr/lib/system/libsystem_networkextension.dylib
    0x7fff7a310000 -     0x7fff7a317fff  libsystem_notify.dylib (172.200.21) <145B5CFC-CF73-33CE-BD3D-E8DDE268FFDE> /usr/lib/system/libsystem_notify.dylib
    0x7fff7a318000 -     0x7fff7a321fef  libsystem_platform.dylib (177.250.1) <16949870-9152-3B0D-9EF0-40FAF84B0F1E> /usr/lib/system/libsystem_platform.dylib
    0x7fff7a322000 -     0x7fff7a32cff7  libsystem_pthread.dylib (330.250.2) <2D5C08FF-484F-3D59-9132-CE1DCB3F76D7> /usr/lib/system/libsystem_pthread.dylib
    0x7fff7a32d000 -     0x7fff7a330ff7  libsystem_sandbox.dylib (851.260.2) <9CACC5F5-3859-3A1F-BCC6-96DDD4A556A8> /usr/lib/system/libsystem_sandbox.dylib
    0x7fff7a331000 -     0x7fff7a333ff3  libsystem_secinit.dylib (30.260.2) <88F3CE61-5FAD-3CDA-AA13-B5B18B3EBD26> /usr/lib/system/libsystem_secinit.dylib
    0x7fff7a334000 -     0x7fff7a33bff3  libsystem_symptoms.dylib (820.267.1) <03F1C2DD-0F5A-3D9D-88F6-B26C0F94EB52> /usr/lib/system/libsystem_symptoms.dylib
    0x7fff7a33c000 -     0x7fff7a351fff  libsystem_trace.dylib (906.260.1) <100013AE-0443-3CF0-AC17-5D934608B60B> /usr/lib/system/libsystem_trace.dylib
    0x7fff7a353000 -     0x7fff7a358ffb  libunwind.dylib (35.4) <24A97A67-F017-3CFC-B0D0-6BD0224B1336> /usr/lib/system/libunwind.dylib
    0x7fff7a359000 -     0x7fff7a388fff  libxpc.dylib (1336.261.2) <9A0FFA79-082F-3293-BF49-63976B073B74> /usr/lib/system/libxpc.dylib
```

In my build-all.py I have following dependencies:

```
PYTHON_VERSIONS=["3.7.3"]
JSON_VERSION="v3.6.1"
OCE_VERSION="0.18"
# OCCT_VERSION="7.1.0"
# OCCT_HASH="89aebde"
# OCCT_VERSION="7.2.0"
# OCCT_HASH="88af392"
OCCT_VERSION="7.3.0p3"
BOOST_VERSION="1.59.0"
#PCRE_VERSION="8.39"
PCRE_VERSION="8.43"
#LIBXML2_VERSION="2.9.3"
LIBXML2_VERSION="2.9.9"
#CMAKE_VERSION="3.4.1"
CMAKE_VERSION="3.14.5"
SWIG_VERSION="3.0.12"
#SWIG_VERSION="4.0.0"
#OPENCOLLADA_VERSION="v1.6.63"
OPENCOLLADA_VERSION="v1.6.68"
```

which is also visible here:

<img width="413" alt="Screenshot 2019-07-09 08 03 14" src="https://user-images.githubusercontent.com/2762514/60863176-1d8ddb80-a220-11e9-8603-5d806055c5fb.png">

Have you tested it with Python 3.7.3 on other OS (Win 10, Linux)? I would like to understand if the v0.6.0 can be successfully built elsewhere.  I have tried it on MSYS2 and Win10 but I am fighting with other problems there.

**Comment by aothms** [comment]
Yes, our IfcOpenBot has built everything just fine, including OS X https://github.com/IfcOpenBot/IfcOpenShell/commit/b4ce5be86d293246dd84f59891e2d5d7f2240284#commitcomment-34192174 It's not a very recent OS X, not entirely sure which version.

> I tested it also with the swig 3.0.12 and the build process resulted in the same error.

Oh sorry, I meant compiling the basic swig example with 3.0.12.

**Comment by atibaut** [comment]
@aothms (cc: @Sara-GO) I have tested it also with the swig 3.0.12, see:

```
$ swig -version
SWIG Version 3.0.12
Compiled with clang++ [x86_64-apple-darwin18.6.0]
Configured options: +pcre

```
and the test passed:

```
$ python3
Python 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 16:52:21) 
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import example
>>> example.fact(10)
3628800
>>> example.my_mod(14,4)
2
>>> example.get_time()
'Mon Jul 15 23:37:38 2019\n'

```
So we can disregard the swig as the source of error.

Never give up!

**Comment by aothms** [comment]
> Never give up!

I like that.

It really seems it's a common problem though. Just googling `macOS python 3.7 segmentation fault` gives tons of results.

For me it's really hard to debug this though.

There are specific things related to macOS in the build script, like https://github.com/IfcOpenShell/IfcOpenShell/blob/v0.6.0/nix/build-all.py#L728 and the `undefined suppress flat_namespace` thing (also in the example you used) is a bit of a hack. There is a better way to ask the Python interpreter what compilation flags to use https://github.com/DLR-SC/tigl-conda/blob/master/python-occ_0.17/link_python.patch It might be better that way and perhaps can resolve this issue. Another thing is that the build script downloads and compiles a whole bunch of pythons. It might be better to (have a setting to) just link and compile against the system python.

**Comment by atibaut** [comment]
Thanks @aothms, we (cc: @Sara-GO) will look at that. Although, I we have already looked at many segfault reports on the web.
Since there is the same behaviour for Python < 3.7.3 (I tested it with some of them on the list in the build script), I also starting to think that this might be a misconfiguration on my system (like clang version...).
Will have closer look to the links you found.

**Comment by atibaut** [comment]
Lucky week, lucky day! I solved the problem.  
The key direction was your last sentence

> It might be better to (have a setting to) just link and compile against the system python. 

Besides the use of system installed python in the build-all.py script, I also did some other small improvements (also inspired by joint research with @Sara-GO) that finally resulted in successful build process for Mac OS X (in my case macOS Mojave version 10.14.5) and Python 3.7.3.

Later on I will create a new pull request for the build-all.py script so everyone can enjoy its use.

Thanks @aothms for your advices!

This is my terminal output:

```
$ python3 build-all.py
This script fetches and builds IfcOpenShell and its dependencies

Script configuration:

* Target Architecture    = x86_64
 - Whether 32-bit (i686) or 64-bit (x86_64) will be built.
* USE_OCCT               = True
 - Compiling against official Open Cascade
* Dependency Directory   = /Users/andrejt/Research/repositories/git/IfcOpenShell/build/Darwin/18.6.0/x86_64
 - The directory where IfcOpenShell dependencies are installed.
* Build Config Type      = RelWithDebInfo
 - The used build configuration type for the dependencies.
   Defaults to RelWithDebInfo if not specified.
* IFCOS_NUM_BUILD_PROCS  = 9
 - How many compiler processes may be run in parallel.

Building: json boost libxml2 python pcre swig occ OpenCOLLADA IfcParse IfcGeom IfcGeomServer IfcOpenShell-Python IfcConvert
using command log file '/Users/andrejt/Research/repositories/git/IfcOpenShell/build/Darwin/18.6.0/x86_64/logs/20190720.log'
Collecting dependencies:
Found existing cmake-3.14.5, skipping
Found existing pcre-8.43, skipping
Found existing swig, skipping
Found existing occt-7.3.0p3, skipping
Found existing libxml2-2.9.9, skipping
Found existing OpenCOLLADA, skipping
Found existing python-3.7.3, skipping
Found existing boost-1.59.0, skipping
Building IfcOpenShell:
Configuring executables...
Building executables...   
Configuring python 3.7.3 wrapper...
* PYTHON_LIBRARY = /Library/Frameworks/Python.framework/Versions/3.7/lib/libpython*.*
* PYTHON_INCLUDE = /Library/Frameworks/Python.framework/Versions/3.7/include/python3.7m
* PYTHON_EXECUTABLE = /Library/Frameworks/Python.framework/Versions/3.7/bin/python3
Building python 3.7.3 wrapper...
* IFCOS built dir = /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/ifcopenshell
* IFCOS archive dir = /Users/andrejt/Research/repositories/git/IfcOpenShell/build/Darwin/18.6.0/x86_64/install/ifcopenshell/python-3.7.3
Built IfcOpenShell...
```

**Comment by aothms** [comment]
Wonderful, thanks both for your continued effort! Using System python is really a nice feature.

**Comment by atibaut** [comment]
This is the console output with the latest commit:

```
$ python3 build-all.py
This script fetches and builds IfcOpenShell and its dependencies

Script configuration:

* Target Architecture    = x86_64
 - Whether 32-bit (i686) or 64-bit (x86_64) will be built.
* USE_OCCT               = True
 - Compiling against official Open Cascade
* Dependency Directory   = /Users/andrejt/Research/repositories/git/IfcOpenShell/build/Darwin/18.6.0/x86_64
 - The directory where IfcOpenShell dependencies are installed.
* Build Config Type      = RelWithDebInfo
 - The used build configuration type for the dependencies.
   Defaults to RelWithDebInfo if not specified.
* IFCOS_NUM_BUILD_PROCS  = 9
 - How many compiler processes may be run in parallel.

Building: boost libxml2 json occ python swig pcre OpenCOLLADA IfcParse IfcGeom IfcGeomServer IfcOpenShell-Python IfcConvert
using command log file '/Users/andrejt/Research/repositories/git/IfcOpenShell/build/Darwin/18.6.0/x86_64/logs/20190721.log'
Collecting dependencies:
Found existing cmake-3.14.5, skipping
Found existing pcre-8.43, skipping
Found existing swig, skipping
Found existing occt-7.3.0p3, skipping
Found existing libxml2-2.9.9, skipping
Found existing OpenCOLLADA, skipping
Found existing python-3.7.3, skipping
Found existing boost-1.59.0, skipping
Building IfcOpenShell:
Configuring executables...
Building executables...
Configuring python 3.7.3 wrapper...
System python version 3.7.3 used: 
PYTHON_EXECUTABLE = /Library/Frameworks/Python.framework/Versions/3.7/bin/python3.7
PYTHON_LIBRARY = /Library/Frameworks/Python.framework/Versions/3.7/lib/libpython*.*
PYTHON_INCLUDE = /Library/Frameworks/Python.framework/Versions/3.7/include/python3.7m
Building python 3.7.3 wrapper...
IfcOpenShell package built to: /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/ifcopenshell
IfcOpenShell package copied to: /Users/andrejt/Research/repositories/git/IfcOpenShell/build/Darwin/18.6.0/x86_64/install/ifcopenshell/python-3.7.3
Built IfcOpenShell...

```

**Comment by BIMvoice** [comment]
Recommend closing as obsolete: this was a `build-all.py` segfault on macOS from 2019. That build path has been superseded (current docs build via CMake, and official prebuilt wheels/binaries cover macOS including arm64), and the report had no follow-up since. Anyone hitting build issues today would be on an entirely different toolchain.

**Comment by BIMvoice** [comment]
Closing per the note above: this was resolved by the reporter in this same thread back in 2019 (switching to system Python instead of the build script's downloaded Python fixed the segfault), and it's been two weeks since the obsolescence note with no objection. Reopen if this build path is still relevant for anyone.

This comment was written with AI assistance.

**Comment by atibaut** [comment]
Hi @BIMvoice, thanks for closing the historical issue.
